lib/spack/spack/database.py

```diff
@@ -698,7 +706,7 @@ def query_one(self, query_spec, known=any, installed=True):
         query. Returns None if no installed package matches.
 
         """
-        concrete_specs = self.query(query_spec, known, installed)
+        concrete_specs = self.query(query_spec, known, installed, unique=True)
         assert len(concrete_specs) <= 1
```

Comment:
I'm not particularly attached to this solution, it's only the first one that did what I wanted. That being said, I find the concept of ```query_one``` strange, imho it should just be a parameter.

Nevertheless, if we can accomplish the load-fix in ```module.py``` I'm all for it. Shall I try there again? Discussions here suggested that, a check of ```specs = args.specs(**query_args)``` might be sufficient.
I'll try that later (or tomorrow). I haven't done it so far, as I wanted some Feedback on how you want this PR done first.